app/src/main/java/com/rizadwi/snapsift/datasource/repository/impl/SourceRepositoryImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.rizadwi.snapsift.datasource.repository.impl
+
+import com.rizadwi.snapsift.common.wrapper.Result
+import com.rizadwi.snapsift.datasource.repository.SourceRepository
+import com.rizadwi.snapsift.datasource.service.NewsService
+import com.rizadwi.snapsift.model.Source
+import com.rizadwi.snapsift.util.extension.getResult
+import javax.inject.Inject
+
+class SourceRepositoryImpl @Inject constructor(private val service: NewsService) :
+    SourceRepository {
+    override suspend fun getByCategory(category: String): Result<List<Source>> {
+        return try {
+            val response = service.getNewsSourcesByCategory(category)
+            when (val it = response.getResult()) {
```

Comment:
what's the meaning of it?

Reply:
`service.getNewsSourcesByCategory(category)` is a normal retrofit interface to get the data from API. it returns the data, but wrapped Result class wrapper by Retrofit package. 

so to get the actual data inside the wrapper we typically do some successful checking & get the underlying data like this:
```
if (result.isSuccessful) {
  val data = result.body()
} else {
 // handle unsuccessful
}
```

but since it's kind of repetitive to do that repeatedly, i create an extension function `getResult()` for that to do the successful checking, extract the result, handle when unsuccessful. please [check this out](https://github.com/rizadwiandhika/snap-sift/blob/d9967f39c7a142e78867a8b679220a85359ffe4f/app/src/main/java/com/rizadwi/snapsift/util/extension/Retrofit.kt#L8)